flocker/volume/filesystems/zfs.py

```diff
@@ -67,26 +71,100 @@ def connectionLost(self, reason):
         del self._result
 
 
-def zfs_command(reactor, arguments):
+class _AsyncLZC(object):
+    def __init__(self, reactor):
+        self._reactor = reactor
+        self._cache = {}
+
+    def callDeferred(self, func, *args, **kwargs):
```

Comment:
It's actually not private and is used in several places. The missing documentation would help, of course.
It turned out that in some cases it was insufficient to call a single `lzc_` function in the asynchronous fashion.
For example, `_list_snapshots` calls `_do_list_snapshots` via `callDeferred`.
Perhaps there was a better way to do that...